lib/src/bazelify/initialize.dart

```diff
@@ -186,80 +187,167 @@ class _GitTagRulesSource implements DartRulesSource {
-  Directory.current = previousCurrent;
-
-  // Read the package's pubspec and the generated .packages file.
-  final pubspec = await Pubspec.fromPackageDir(arguments.pubPackageDir);
-  final packagesFilePath = p.join(arguments.pubPackageDir, '.packages');
-  final packages = parse(
-    await new File(packagesFilePath).readAsBytes(),
-    Uri.parse(packagesFilePath),
-  );
-
-  // Clean and re-build the .bazelify folder.
-  final bazelifyPath = p.join(arguments.pubPackageDir, '.bazelify');
-  final bazelifyDir = new Directory(bazelifyPath);
-  if (await bazelifyDir.exists()) {
-    await bazelifyDir.delete(recursive: true);
+class _Initialize {
+  final BazelifyInitArguments arguments;
+
+  _Initialize(this.arguments);
+
+  Future<Null> run() async {
+    final timings = <String, Duration>{};
+    final stopwatch = new Stopwatch()..start();
+    await _pubGetInPackage();
+    timings['pub get'] = stopwatch.elapsed;
+    stopwatch.reset();
+    final buildFilePaths = await _createBazelifyDir();
+    timings['create .bazelify'] = stopwatch.elapsed;
+    stopwatch.reset();
+    await _writeBazelFiles(buildFilePaths);
+    timings['create packages.bzl, build, and workspace'] = stopwatch.elapsed;
+    stopwatch.reset();
+    await _suggestAnalyzerExcludes();
+    timings['scan for analysis options'] = stopwatch.elapsed;
+    _printTiming(timings);
+  }
+
+  Future<Null> _pubGetInPackage() async {
+    final previousDirectory = Directory.current;
+    Directory.current = new Directory(arguments.pubPackageDir);
+    await Process.run(arguments.pubExecutable, const ['get']);
+    Directory.current = previousDirectory;
+  }
+
+  Future<Map<String, String>> _createBazelifyDir() async {
+    final packages = await _readPackages();
+    final bazelifyPath = p.join(arguments.pubPackageDir, '.bazelify');
+    await _createEmptyDir(bazelifyPath);
+    final buildFilePaths =
+        await _writePackageBuildFiles(bazelifyPath, packages);
+    return buildFilePaths;
+  }
+
+  Future<Null> _writeBazelFiles(Map<String, String> buildFilePaths) async {
+    await _writePackagesBzl(buildFilePaths);
+    await _writeWorkspaceFile();
+    await _writeBuildFile();
   }
-  await bazelifyDir.create(recursive: true);
-
-  // Store the current path.
-  final packageToPath = <String, String>{};
-  for (final package in packages.keys) {
-    // Get ready to create a <name>.BUILD.
-    final buildFilePath = p.join(bazelifyPath, '$package.BUILD');
-    var localPath = packages[package].toFilePath();
-    localPath = localPath.substring(0, localPath.length - 'lib/'.length);
-    packageToPath[package] = localPath;
-
-    // Create a new build file for this directory and write to disk.
-    final newBuildFile = await BuildFile.fromPackageDir(localPath);
-    await new File(buildFilePath).writeAsString(newBuildFile.toString());
+
+  Future<Map<String, Uri>> _readPackages() async {
+    final packagesFilePath = p.join(arguments.pubPackageDir, '.packages');
+    return parse(
```

Comment:
Not really related to this cl, but imo this library should be imported with a prefix.